tests/multi_threaded.rs

```diff
@@ -0,0 +1,146 @@
+    // needed.
+    {
+        let data_subscriber_1 = event_manager.subscriber_mut(ds_id_1).unwrap();
+        data_subscriber_1.trigger_all_counters();
+
+        let data_subscriber_2 = event_manager.subscriber_mut(ds_id_2).unwrap();
+        data_subscriber_2.trigger_all_counters();
+    }
+
+    let thread_handle = thread::spawn(move || {
+        // We cannot run the event manager loop for ever. Let's just run it for 100 iterations.
+        for _ in 0..100 {
+            // When the event manager loop exits, it will return the number of events it
+            // handled.
+            let ev_count = event_manager.run().unwrap();
+            // Since we triggered all counters, we're expecting the number of events to always be
+            // 6 (3 from one subscriber and 3 from the other).
+            assert_eq!(ev_count, 6);
+        }
+
+        // One really important thing is to always remove the subscribers once you don't need them
+        // any more.
+        // When calling remove_subscriber, you receive ownership of the subscriber object.
+        let data_subscriber_1 = event_manager.remove_subscriber(ds_id_1).unwrap();
+        let data_subscriber_2 = event_manager.remove_subscriber(ds_id_2).unwrap();
+
+        // Let's check how the subscribers look after 100 iterations.
+        // Each subscriber's counter start from 0. When an event is triggered, each counter
+        // is incremented. So after 100 iterations when they're all triggered, we expect them
+        // to be 100.
+        let expected_subscriber_counters = vec![100, 100, 100];
+        assert_eq!(
+            data_subscriber_1.get_all_counter_values(),
+            expected_subscriber_counters
+        );
+        assert_eq!(
+            data_subscriber_2.get_all_counter_values(),
+            expected_subscriber_counters
+        );
+    });
+
+    thread_handle.join().unwrap();
+}
+
+// Showcase how you can manage subscribers that make use of inner mutability using the event manager
+// in a multithreaded context.
+#[test]
+fn test_subsscriber_inner_mut() {
+    // We are using just the `CounterInnerMutSubscriber` subscriber, so we could've initialize
+    // the event manager as: EventManager::<CounterInnerMutSubscriber>::new().
+    // The typical application will have more than one subscriber type, so let's just pretend
+    // this is the case in this example as well.
+    let mut event_manager = EventManager::<Arc<dyn EventSubscriber + Send + Sync>>::new()
+        .expect("Cannot create event manager");
+
+    let subscriber = Arc::new(CounterInnerMutSubscriber::new());
+    subscriber.trigger_event();
+
+    // Let's just clone the subscriber before adding it to the event manager. This will allow us
+    // to use it from this thread without the need to call into EventManager::subscriber_mut().
+    let subscr_id = event_manager.add_subscriber(subscriber.clone());
```

Comment:
Not really important, but `subscr_id` sounds jarring somehow :-s How about `sub_id`? :-s

Reply:
I really avoided sub_id on purpose. `sub` for me means substitution, not subscriber. I'll just name it `subscriber_id`. I will assume humans have big enough screens nowadays to not be bother by a 13 chars long name :D